app/src/main/java/protect/card_locker/preferences/SettingsActivity.java

```diff
@@ -140,6 +142,29 @@ public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
                 refreshActivity(true);
                 return true;
             });
+
+            ListPreference colorPreference = findPreference(getResources().getString(R.string.setting_key_theme_color));
+            assert colorPreference != null;
+            colorPreference.setOnPreferenceChangeListener((preference, o) -> {
+                refreshActivity(true);
+                return true;
+            });
+            if (!DynamicColors.isDynamicColorAvailable()) {
+                // remove catima from list
+                CharSequence[] colorValues = colorPreference.getEntryValues();
+                CharSequence[] colorEntries = colorPreference.getEntries();
+                CharSequence[] newColorValues = new CharSequence[colorValues.length - 1];
+                CharSequence[] newColorEntries = new CharSequence[colorEntries.length - 1];
+                for (int i = 0, j = 0; i < colorValues.length; i++) {
+                    if (!colorValues[i].equals(getResources().getString(R.string.settings_key_catima_theme))) {
+                        newColorValues[j] = colorValues[i];
+                        newColorEntries[j] = colorEntries[i];
+                        j++;
+                    }
+                }
+                colorPreference.setEntries(newColorEntries);
+                colorPreference.setEntryValues(newColorValues);
+            }
```

Comment:
Hmm, given we're looping over everything here... maybe we could rename "System" to "Catima" on older Android versions because "System" makes less sense? In that case though this code can very easily become quite complex so maybe we should just have 2 different arrays and do something like this?

```
if (!DynamicColors.isDynamicColorAvailable()) {
    colorPreference.setEntries(R.array.color_value_strings_no_dynamic);
    colorPreference.setEntryValues(R.array.color_values_no_dynamic);
}
```

I'm not quite sure what to do here, I can't think of a way that isn't smelly, but I'm thinking having 2 arrays, while more repetitive, keeps code more readable? :/

The reason all of this sucks is because the default needs to be `System` on Android 12+ but `Catima` before that, as `System` doesn't exist back then.